### PR TITLE
Fix EntryPointBuilder when no GET operation

### PR DESCRIPTION
**Pull request by ogizanagi**
Following #205 

Sometimes, it might happen to have resources without any GET collection operation. However, those resources might have a POST operation in order to create a new item or submit a payload.

This is currently impossible, as the resource IRI is generated from the first GET collection operation.
However, this fix is more a workaround than a real solution.

On another hand, I easily understand why the resource and item IRIs are related to the Route, but doesn't this system have some limitations as it is ?
A resource only exposing GET item operation but not a GET collection operation shouldn't being part of the EntryPoint anyway ? 
The POST collection operation might be available, we'll see it under the Entrypoint `hydra:supportedProperty` on the hydra `apidoc`, but not on the `/api` entrypoint ?

Would always generating a HEAD collection operation & route for resources be a good alternative to define the resource IRI, or is it a non-sense ? (the HEAD operation behavior should of course remain customizable).

### CODE REVIEW

**Comment by dunglas** [comment]
IMO having a GET collection route is mandatory. Even if this route only return a description or an empty JSON-LD document. This is implied by how REST is designed.

**Comment by ogizanagi** [comment]
Well, it might be mandatory in order to expose it in the API entrypoint. But I don't think it is prohibited to not have a GET collection route for resources that don't need one.

**Comment by dunglas** [comment]
If doesn't exist, in term of HTTP the URI `http://example.com/mycollection` doesn't exist either.

**Comment by ogizanagi** [comment]
I'm curious about how you deal with simple services like workflow actions ?
Our approach was to represent the needed payload as everything in the API, as a resource, but only with a POST collection operation.

If any resource should define a GET collection operation, then, I guess we should generate one by default if not provided, only exposing the hydra vocab & supported operations on it, instead of letting the api entrypoint crash.

File: JsonLd/EntrypointBuilder.php [hunk 1]
```diff
@@ -59,8 +60,17 @@ public function getEntrypoint()
         ];
 
         foreach ($this->resourceCollection as $resource) {
-            if (!empty($resource->getCollectionOperations())) {
-                $entrypoint[lcfirst($resource->getShortName())] = $this->iriConverter->getIriFromResource($resource);
+            if (!empty($operations = $resource->getCollectionOperations())) {
+                try {
+                    $entrypoint[lcfirst($resource->getShortName())] = $this->iriConverter->getIriFromResource($resource);
+                } catch (InvalidArgumentException $ex) {
+                    foreach ($operations as $operation) {
```
Comment (sroze) [inline JsonLd/EntrypointBuilder.php]: Please move this check into a `hasGetOperation` method or similar.

**Comment by sroze** [comment]
@dunglas I think that a GET route shouldn't be mandatory. The HTTP verb and the URI are not dependent each others, the verb is just the meaning of the action of this collection, a write-only collection might be extremely useful, especially in @ogizanagi's case (a workflow).

**Comment by dunglas** [comment]
Ok LGTM. @ogizanagi can you just do the small changes asked by @sroze?

**Comment by ogizanagi** [comment]
@dunglas : Done. (Unless you prefer having it into the `ResourceInterface`, but I don't think it is worth it at all).

**Comment by sroze** [comment]
That's perfect, thanks @ogizanagi ! :+1:

**Comment by dunglas** [comment]
@sroze it should be merged in 1.0.

**Comment by sroze** [comment]
Erf I thought it was, didn't check. :/ We need to improve this contribution workflow btw.

**Pull request merged**